src/communication/network_interface_sr.c

```diff
@@ -7281,27 +7281,40 @@ slogtb_dump_trantable (THREAD_ENTRY * thread_p, unsigned int rid, char *request,
 void
 smmon_get_server_info (THREAD_ENTRY * thread_p, unsigned int rid, char *request, int reqlen)
 {
-  char *buffer = NULL, *ptr;
+  char *buffer_a = NULL, *ptr, *buffer;
   int size = 0;
   OR_ALIGNED_BUF (OR_INT_SIZE + OR_INT_SIZE) a_reply;
   char *reply = OR_ALIGNED_BUF_START (a_reply);
   int error = NO_ERROR;
   MMON_SERVER_INFO server_info;
+  int string_len;
 
   mmon_aggregate_server_info (server_info);
 
-  size += or_packed_string_length (server_info.name, NULL);
+  string_len = or_packed_string_length (server_info.name, NULL);
+  if (string_len % MAX_ALIGNMENT)
+    {
+      size += string_len + OR_INT_SIZE;
+    }
+  else
+    {
+      size += string_len;
+    }
+  //size += or_packed_string_length (server_info.name, NULL);
   // Size of total_mem_usage
   size += OR_INT64_SIZE;
 
-  buffer = (char *) db_private_alloc (thread_p, size);
-  if (buffer == NULL)
+  buffer_a = (char *) db_private_alloc (thread_p, size + MAX_ALIGNMENT);
```

Comment:
You can remove the previous complex things to make alignment. 'size + MAX_ALIGNMENT' covers enough.

Reply:
I can't understand it. What is "the previous complex things"? Please explain more.

Reply:
calculation for the alignment.

Reply:
you mean the line `size = size % MAX_ALIGNMENT ? size + INT_ALIGNMENT : size;`?

Reply:
No. You just use size + MAX_ALIGNMENT.